### ISSUE TITLE
Flesh out multitenancy

### ISSUE THREAD

**Issue by mhuang-usdr** [usdigitalresponse/usdr-gost#72]
### **Tenant-scoped**

_Definition_: grab all agencies assigned to a tenant, for example children, grandchildren AND parents, grandparents, AND cousins, etc. Everything belonging to a tenant

_Places in the UI that should be tenant-scoped are:_

- [x] Users
- [x] Agencies (see #11 for more bg info)
- [x] Profile settings > change agency
- [x] Browse Grants > Viewed By
- [x] Browse Grants > Interested Agencies
- [x] Click into any grant modal > Interested Agencies
- [x] Click into any grant modal > Assigned Agencies
- [x] Click into any grant modal > Assigned Agencies dropdown (a la #61 )

### **Agency scoped** 

_Definition:_ the agency only. No children or grandchildren either, just the self.

Places in the UI that should be agency-scoped are:

- [x] Dashboard - All the grey boxes (Total Grants Matching Search Criteria, New Grants Matching, Updated Grants Matching Search Criteria, Total Viewed Grants, Total Interested Grants)
- [x] Keywords
- [x] Eligibility codes
- [x] My Grants > Interested
- [x] My Grants > Assigned

### **Agency + child scoped** 

_Definition:_ the agency + children and grandchildren.

Places in the UI that should be agency+child-scoped are:

- [x] Dashboard - Total Interested Grants by Agencies, Recent Activity, Upcoming Closing Dates
- [x] Agency View Switcher - Profile picture > Settings

**Comment by bspates** [comment]
Copying over discussion from slack:
```
Brian:
Similar question to earlier, but slightly different context. For displaying users on the users tab, right now it is scoped such that an admin user can only see other users at their agency level or below. This should now be tenant wide as well?

Mindy: 
Tl;dr - same as the dropdowns - let's scope such that all users can see everyone in the tenant, but only admins can edit users that lie within their agencies or subagencies.My thought process below:

    The concerns of seeing all users in a tenant is information overload.

    Right now we have the ability to sort by agency, so while the situation is not ideal, the overload isn't as bad as just a straight list.

    The pros of seeing all users would help in breaking down information silos - allows users to help pinpoint who's working on what  

    I've asked govts before whether they want to keep their activity secret from other agencies, and the answer is a NO - their problem is always that they don't have enough visibility

    Thus, the ideal solution is probably for all users to be able to see everyone, and also introduce a better search/sort/filter to the tables down the road.
    We can validate this with NV - ask when we see them next Tuesday.

```

**Comment by bspates** [comment]
@mhuang-usdr Going through the workflows and the code. There is these repeated concept of which Agency you currently are. This can be selected from the profile settings of the user. Due to the changes in visibility between agencies in the same tenant we discussed, I am wondering if this concept is still needed. It seems like it was mostly to enforce the agency/sub-agency access scoping.

**Comment by mhuang-usdr** [comment]
The impersonation feature is still needed for a few cases - an admin in a
parent agency needs to update the keywords or eligibility codes for a child
agency. Or an admin of a parent agency wants to see the to-do list of a
child agency.

This is useful, for example, if someone is OOO and another person needs to
step in. Or if USDR, or another admin, needs to help another agency
troubleshoot.

That being said it is not a common UX pattern, and I am very open to other
ways to solve the user requirements. But until we have a better solution
I'd keep this feature in, unless it's causing blocking issues on the eng
side.


Mindy

**Comment by mhuang-usdr** [comment]
@bspates ^ fyi

**Comment by bspates** [comment]
@mhuang-usdr Should I keep the scope of agency impersonation limited to subagencies or should it be open to all agencies in the tenant now?

**Comment by mhuang-usdr** [comment]
@bspates that should be kept to subagencies

**Comment by mhuang-usdr** [comment]
QA'd in prod
